### PR TITLE
[Typography] Add unit test to improve coverage

### PR DESCRIPTION
**Pull request by ianegordon**

### CODE REVIEW

**Comment by jverkoey** [comment]
```
ERROR: /Volumes/BuildData/tmpfs/src/github/repo/components/Typography/BUILD:34:1: C++ compilation of rule '//components/Typography:unit_test_sources' failed (Exit 1)
components/Typography/tests/unit/TypographyTests.m:20:9: fatal error: '../../src/private/UIFont+MaterialTypographyPrivate.h' file not found
#import "../../src/private/UIFont+MaterialTypographyPrivate.h"
        ^
1 error generated.
```

**Comment by jverkoey** [comment]
I believe there have been patterns of using private headers in tests before - check for other components that have imported similar private headers.

**Comment by romoore** [comment]
Yes, this is a general pattern we use in exposing private classes and APIs for testing.


```bash

rsmoore:material-components-ios rsmoore$ find . -iname BUILD | xargs grep "name = \"private"
./components/Tabs/BUILD:    name = "private",
./components/MaskedTransition/BUILD:    name = "private",
./components/FeatureHighlight/BUILD:    name = "private",
./components/AppBar/BUILD:    name = "private",
./components/Ink/BUILD:    name = "private",
./components/Buttons/BUILD:    name = "private",
./components/ButtonBar/BUILD:    name = "private",
./components/TextFields/BUILD:    name = "private",
./components/Chips/BUILD:    name = "private",
./components/Snackbar/BUILD:    name = "private",
./components/BottomAppBar/BUILD:    name = "private",
./components/Slider/BUILD:    name = "private",
./components/private/Overlay/BUILD:    name = "private",
./components/private/ThumbTrack/BUILD:    name = "private_headers",
./components/Dialogs/BUILD:    name = "private",
./components/BottomNavigation/BUILD:    name = "private",
./components/PageControl/BUILD:    name = "private",
./components/Collections/BUILD:    name = "private",
./components/ProgressView/BUILD:    name = "private",

```

File: components/Typography/tests/unit/TypographyTests.m [hunk 1]
```diff
@@ -366,4 +367,25 @@ - (void)testFontFamilyMatchesSystemFontFamily {
   }
 }
 
+- (void)testExtendedDescription {
+  // Given
+  UIFont *systemFont;
+  if ([UIFont respondsToSelector:@selector(systemFontOfSize:weight:)]) {
+#pragma clang diagnostic push
```
Comment (romoore) [inline components/Typography/tests/unit/TypographyTests.m:374]: Can we add a TODO to track removing this?

**Review by romoore — APPROVED** [review]

**Pull request merged**